source/OpenBveApi/Train/AbstractTrain.cs

```diff
@@ -39,5 +39,11 @@ public virtual void Derail(int CarIndex, double ElapsedTime)
 		public virtual void Dispose()
 		{
 		}
+
+		/// <summary>Stops all sounds from this train</summary>
+		public virtual void StopAllSounds()
+		{
+
+		}
 	}
```

Comment:
```csharp
public AbstractCar[] Cars;
```
It may be better to move Cars from the Train class of each program.
Then you can eliminate the override of the StopAllSounds function.